alica_tests/src/test/test_alica_behaviourtrigger.cpp

```diff
@@ -109,15 +111,15 @@ TEST_F(AlicaBehaviourTrigger, triggerTest) {
     }
     alicaTests::TestWorldModel::getOne()->trigger1->run();
     alicaTests::TestWorldModel::getOne()->trigger2->run();
-    this_thread::sleep_for(duration);
+    step(ae);
```

Comment:
This is not the step you want.
BehaviourTrigger are independent of engine steps, they intentionally run asynchronously.

The Improve Query PR has some changes to this test.

Reply:
I'll allow the Query PR to be merged so I can incorporate those changes into here before addressing this.